src/drm.cpp

```diff
@@ -1279,22 +1288,56 @@ void drm_unlock_fbid( struct drm_t *drm, uint32_t fbid )
 }
 
 /* Handle the orientation of the display */
-uint64_t get_drm_effective_orientation()
+	void update_drm_effective_orientation(struct drm_t *drm,  struct connector *conn)
 {
-	switch ( g_drmModeOrientation )
+	drm_screen_type screenType = drm_get_screen_type(drm);
+	if ( screenType == DRM_SCREEN_TYPE_INTERNAL )
+	{		
+		switch ( g_drmModeOrientation )
+		{
+			case PANEL_ORIENTATION_0:
+				g_drmEffectiveOrientation = DRM_MODE_ROTATE_0;
+				break;
+			case PANEL_ORIENTATION_90:
+				g_drmEffectiveOrientation = DRM_MODE_ROTATE_90;
+				break;
+			case PANEL_ORIENTATION_180:
+				g_drmEffectiveOrientation = DRM_MODE_ROTATE_180;
+				break;
+			case PANEL_ORIENTATION_270:
+				g_drmEffectiveOrientation = DRM_MODE_ROTATE_270;
+				break;
+			case PANEL_ORIENTATION_AUTO:
+				if (conn->props.count("panel orientation") > 0 ) 
+				{
+					const char *orientation = get_enum_name(conn->props["panel orientation"], conn->initial_prop_values["panel orientation"]);
+
+					if (strcmp(orientation, "Normal") == 0)
+					{
+						g_drmEffectiveOrientation = DRM_MODE_ROTATE_0;
+					}
+					else if (strcmp(orientation, "Left Side Up") == 0)
+					{
+						g_drmEffectiveOrientation = DRM_MODE_ROTATE_90;
+					}
+					else if (strcmp(orientation, "Upside Down") == 0)
+					{
+						g_drmEffectiveOrientation = DRM_MODE_ROTATE_180;
+					}
+					else if (strcmp(orientation, "Right Side Up") == 0)
+					{
+						g_drmEffectiveOrientation = DRM_MODE_ROTATE_270;
+					}
+					break;
+				}
+				g_drmEffectiveOrientation = g_bRotated ? DRM_MODE_ROTATE_270 : DRM_MODE_ROTATE_0;
```

Comment:
Nit: can we wrap this in an `else` block and remove the `break` statement above to make this a bit more readable?